### ISSUE TITLE
Diff editor scrolls horizontally when dragging selection vertically

### ISSUE THREAD

**Issue by joaomoreno** [microsoft/vscode#117664]
This bug has probably been around forever, funny how it doesn't show up often.

![recording (4)](https://user-images.githubusercontent.com/22350/109131013-36097d00-7707-11eb-8b9c-344dc41da8c6.gif)

**Comment by alexdima** [comment]
@joaomoreno Are you using some screen zooming? (I could not reproduce on Windows)

**Comment by joaomoreno** [comment]
Yeah I know this isn't something which would catch you offguard.

I wonder if the cursor position onthe line above should have the same horizontal offset as the mouse cursor, instead of the end of the line? 🤔

**Comment by alexdima** [comment]
> I wonder if the cursor position onthe line above should have the same horizontal offset as the mouse cursor, instead of the end of the line?

Yes, that would make most sense. The reason it doesn't right now is due to the special case that the line above might not be visible (like if there is a large view zone) and we need a visible line to convert a mouse coordinate -> logical column (due to cases like RTL, bidi, etc.). But in general the line above would be visible so I like your suggestion.
